integrations/MoEngage/browser.js

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable no-undef */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable prefer-template */
+/* eslint-disable yoda */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-sequences */
+/* eslint-disable no-redeclare */
+/* eslint-disable block-scoped-var */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable guard-for-in */
+/* eslint-disable prefer-rest-params */
+/* eslint-disable vars-on-top */
+/* eslint-disable no-var */
+/* eslint-disable no-param-reassign */
+/* eslint-disable func-names */
+import each from "@ndhoule/each";
+import logger from "../../utils/logUtil";
+
+class MoEngage {
```

Comment:
please include comments involving the destination documentation
any special case handling and why it is required and where it's mentioned in destination docs etc

Reply:
done